stan/math/rev/mat/functor/map_rect_concurrent.hpp

```diff
@@ -0,0 +1,102 @@
+#ifndef STAN_MATH_REV_MAT_FUNCTOR_MAP_RECT_CONCURRENT_HPP
```

Comment:
Quick question: why was this logic moved? I didn't see it mentioned in the PR description and it's easier to spot it now in the diff. (Just wondering why it's done this way, I'm not trying to imply that there's anything wrong with this.)

Reply:
It was moved due to dependecy reasons. The threaded `map_rect` depends now on the include

`#include <stan/math/rev/core/chainablestack.hpp>`

as we are sticking the chainablestack instance into the lambda doing all the work. This requires us to move essentially the prim version into rev. This is not ideal, but once we get the TBB we  can handle the instantiation of chainablestack better (for now we have to make sure via the lambda that an instance is available).